forui/test/src/widgets/badge_test.dart

```diff
@@ -0,0 +1,71 @@
+@Tags(['golden'])
+library;
+
+import 'package:flutter/material.dart';
+
+import 'package:flutter_test/flutter_test.dart';
+
+import 'package:forui/forui.dart';
+import '../test_scaffold.dart';
+
+void main() {
+  group('FBadge', () {
+    for (final (name, theme, background) in TestScaffold.themes) {
+      for (final variant in FBadgeVariant.values) {
+        testWidgets('$name with FBadgeContent', (tester) async {
+          await tester.pumpWidget(
+            TestScaffold(
+              data: theme,
+              background: background,
+              child: FBadge(
+                label: 'Badge',
+                design: variant,
+              ),
+            ),
+          );
+
+          await expectLater(
+            find.byType(TestScaffold),
+            matchesGoldenFile('badge/$name-$variant-badge-content.png'),
+          );
+        });
+
+        testWidgets('$name with raw content', (tester) async {
+          await tester.pumpWidget(
+            TestScaffold(
+              data: theme,
+              background: background,
+              child: FBadge.raw(
+                design: variant,
+                builder: (_, style) => Padding(
+                    padding: const EdgeInsets.all(50),
+                    child: Container(
+                      width: 50,
+                      height: 50,
+                      color: Colors.blueAccent,
+                      child: Container(
+                        width: 50,
+                        height: 50,
+                        decoration: BoxDecoration(
```

Comment:
Just curious, could we simplify this? I don't think it's a big deal but I doubt wrapping a container in a container does much.

Reply:
We wrap it in a container because we want to assert that `style.background` is the correct color. However, since the badge's background is the same, it'll be impossible to visually see. We add a blue box surrounding it to be able to be able to see that.

Reply:
Would it make sense to add it directly to the `TestScaffold` since it would probably be a common occurrence.

Reply:
I don't think it's possible since it's something like TestScaffold -> widget under test -> ring -> widget under test